tests/test_form.py

```diff
@@ -46,6 +49,24 @@ def test_multipart_form_data(tmp_path):
             assert response.json() == {"output": file_length**2}
 
 
+def test_file_too_big(tmp_path):
+    file_length = 1024 * 1024 * 100
+
+    server = LitServer(
+        SimpleFileLitAPI(), accelerator="cpu", devices=1, workers_per_device=1, max_payload_size=(file_length / 2)
+    )
+
+    with TestClient(server.app) as client:
+        file_path = f"{tmp_path}/big_file.txt"
+        file_length = 1024 * 1024 * 100
+        with open(file_path, "wb") as f:
+            f.write(bytearray([1] * file_length))
+        with open(file_path, "rb") as f:
+            file = {"input": f}
+            response = client.post("/predict", files=file)
```

Comment:
what happens if a bad actor modifies the content length:

```py
response = client.post("/predict", files=file, headers={"content-length": "1024"})
```

Reply:
Seems like this is open to spoofing.  This seems like an underlying problem with starlette that it does not cease reading when the content-length is reached.  I guess if the user is really worried about it, they should just authenticate their endpoint

Reply:
Hi @aniketmaurya @andyland  , Sorry for interrupting.
Do you think checking the actual body size might help ? 
Perhaps something like this:

```python
body = await request.body()
if len(body) > self.max_size:
    return ...
```

Reply:
that's a good point @bhimrazy! We would need to iteratively read the request data before loading the full data in system memory and raise the MaxSizeError if it exceeds the provided value.

cc: @lantiga

Reply:
yes good point @bhimrazy 
it would add a synchronization point in the async chain, we should compare the effect of @bhimrazy's suggestion on the performance benchmarks we have and if it's negligible include it @aniketmaurya can you make that check?

Reply:
@bhimrazy by reading the body, you are loading it into memory which would defeat the point of the check.

cc @lantiga @aniketmaurya

Reply:
@aniketmaurya you would need to reimplement this function to read it iteratively: https://github.com/encode/starlette/blob/6f863b0d3b8e8f18d5df9e8cd2514f7085b874e1/starlette/requests.py#L221

Reply:
Or we could just switch to this when its merged upstream: https://github.com/encode/starlette/pull/2328/files

Reply:
yes @andyland, overriding the `Request.body` method to check for size iteratively would be the right way to go because as mentioned it defeats the purpose of large data attacks. 

This [Starlette PR](https://github.com/encode/starlette/pull/2328/files) seem to be open for way too long so maybe we should just do it on our side for now.

Reply:
ok updated @aniketmaurya